vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1114,6 +1114,14 @@ def execute_model(
             valid_sampled_token_ids = self.rejection_sampler.parse_output(
                 sampled_token_ids, self.input_batch.vocab_size)
 
+        # Remove sampled tokens from the requests that are still
+        # being prefilled.
+        for i, req_id in enumerate(self.input_batch.req_ids):
+            if (self.requests[req_id].num_computed_tokens +
+                    scheduler_output.num_scheduled_tokens[req_id]
+                    < self.requests[req_id].num_tokens):
+                valid_sampled_token_ids[i] = []
```

Comment:
Can we merge this loop with the one in L1081?

Reply:
I intentionally made this a separate loop first to make sure the overall approach in this PR is on track. I could then try to improve the performance now.